Week3/homework/step2-2.js

```diff
@@ -1,20 +1,26 @@
 'use strict';
 
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
-  const numbers = [];
+  const range = (begin, end) => [...Array(end + 1).keys()].filter(n => n >= begin);
+  const numbers = range(startIndex, stopIndex);
 
-  // Replace this comment and the next line with your code
-  console.log(startIndex, stopIndex, threeCallback, fiveCallback, numbers);
+  numbers.forEach(item => {
+    if (item % 15 === 0) {
+      console.log(item + '=>' + threeCallback(item) + ' and ' + fiveCallback(item));
+    } else if (item % 3 === 0) {
+      console.log(item + '=>' + threeCallback(item));
+    } else if (item % 5 === 0) {
+      console.log(item + '=>' + fiveCallback(item));
+    }
+  });
```

Comment:
The logic of checking the callbacks is correct but am curious to know why you chose to go with the approach of creating `numbers` array with `const range = (begin, end) => [...Array(end + 1).keys()].filter(n => n >= begin);` logic rather than a `for` loop?

Reply:
Maybe because of my bias against `push` method. For creating bigger number arrays I think `push` method in a `for` loop would cost a lot for the process time. That's just my simple assumption. I don't have enough benchmark data to compare it actually 😀